server/src/main/java/org/elasticsearch/cluster/routing/RoutingNodes.java

```diff
@@ -275,7 +275,7 @@ public RoutingNode node(String nodeId) {
 
     public Set<String> getAttributeValues(String attributeName) {
         // Only ever accessed on the master service thread so no need for synchronization
-        assert MasterService.assertMasterUpdateOrTestThread();
+        // assert MasterService.assertMasterUpdateOrTestThread();
```

Comment:
This is now called by `AwarenessAllocationDecider` from a continuous computation that is reusing generic thread pool. 
It also looks like all data used here is immutable and final so I am not sure above comment is relevant and assertion is needed (unless I miss something)